src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -77,28 +77,25 @@ protected Integer initialValue()
         }
     };
 
-
     public TrieMemoryIndex(IndexContext indexContext)
     {
         super(indexContext);
-        //TODO Do we need to follow a setting for this?
-        this.data = new MemtableTrie<>(BufferType.OFF_HEAP);
+        this.data = new MemtableTrie<>(TrieMemtable.BUFFER_TYPE);
         this.primaryKeysReducer = new PrimaryKeysReducer();
-        // MemoryIndex is per-core, so analyzer should be thread-safe..
-        this.analyzerFactory = indexContext.getAnalyzerFactory();
-        this.validator = indexContext.getValidator();
-        this.isLiteral = TypeUtil.isLiteral(validator);
     }
 
-    @Override
-    public long add(DecoratedKey key, Clustering clustering, ByteBuffer value)
+    public void add(DecoratedKey key,
+                    Clustering clustering,
+                    ByteBuffer value,
+                    LongConsumer onHeapAllocationsTracker,
+                    LongConsumer offHeapAllocationsTracker)
     {
         synchronized (writeLock)
```

Comment:
Why even have a separate lock object, when you could just make the whole method `synchronized`?